src/pages/storage/StorageBuckets.tsx

```diff
@@ -0,0 +1,310 @@
+    },
+    {
+      content: SIZE_COL,
+      sortKey: "size",
+      className: "size",
+    },
+    {
+      content: URL_COL,
+    },
+    {
+      content: "",
+      className: "actions u-align--right",
+      "aria-label": "Actions",
+    },
+  ];
+
+  const filteredBuckets = buckets.filter((item) => {
+    if (!filters.queries.every((q) => item.name.toLowerCase().includes(q))) {
+      return false;
+    }
+    if (filters.pools.length > 0 && !filters.pools.includes(item.pool)) {
+      return false;
+    }
+    return true;
+  });
+
+  const rows = filteredBuckets.map((bucket) => {
+    return {
+      key: getBucketKey(bucket),
+      name: getBucketKey(bucket),
+      className: "u-row",
+      columns: [
+        {
+          content: (
+            <div className="u-truncate" title={bucket.name}>
+              {bucket.name}
+            </div>
+          ),
+          role: "rowheader",
+          "aria-label": NAME_COL,
+        },
+        {
+          content: (
+            <ResourceLink
+              type={"bucket"}
+              value={bucket.pool}
+              to={`/ui/project/${project}/storage/pool/${bucket.pool}`}
+            />
+          ),
+          role: "cell",
+          "aria-label": POOL_COL,
+        },
+        {
+          content: bucket.config?.size ?? "-",
+          role: "cell",
+          "aria-label": SIZE_COL,
+        },
+        {
+          content: (
+            <div className="u-truncate" title={bucket.s3_url}>
+              {bucket.s3_url}
+            </div>
+          ),
+          role: "cell",
+          "aria-label": URL_COL,
+        },
+        {
+          className: "actions u-align--right",
+          content: (
+            <StorageBucketsActions
+              bucket={bucket}
+              className="storage-bucket-actions u-no-margin--bottom"
+            />
+          ),
+          role: "cell",
+          "aria-label": ACTIONS_COL,
+        },
+      ],
+      sortData: {
+        name: bucket.name,
+        pool: bucket.pool,
+        size: bucket.config?.size ?? "",
+        s3_url: bucket.s3_url,
+      },
+    };
+  });
+
+  const { rows: sortedRows, updateSort } = useSortTableData({
+    rows,
+    defaultSortDirection: "descending",
+  });
+
+  if (isLoading) {
+    return <Loader isMainComponent />;
+  }
+
+  const hasBuckets = buckets.length !== 0;
+
+  const content = hasBuckets ? (
+    <div className="storage-volumes">
+      <ScrollableTable
+        dependencies={[filteredBuckets]}
+        tableId="bucket-table"
+        belowIds={["status-bar"]}
+      >
+        <TablePagination
+          data={sortedRows}
+          id="pagination"
+          itemName="bucket"
+          className="u-no-margin--top"
+          aria-label="Table pagination control"
+          description={
+            selectedNames.length > 0 && (
+              <SelectedTableNotification
+                totalCount={buckets.length ?? 0}
+                itemName="bucket"
```

Comment:
```suggestion
                itemName="bucket"
                parentName="project"
```
This makes the "all buckets in this project selected", which is more clear than the current "all bucket selected"